core/domain/platform_parameter_domain_test.py

```diff
@@ -81,3 +82,1099 @@ def test_to_dict(self):
+                'user_locale': 'en',
+            },
+            {
+                'server_mode': 'dev',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid browser type \'Invalid\''):
+            context.validate()
+
+    def test_validate_with_invalid_app_version_raises_exception(self):
+        context = parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': 'Android',
+                'browser_type': None,
+                'app_version': 'a.a.a',
+                'user_locale': 'en',
+            },
+            {
+                'server_mode': 'dev',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid version \'a.a.a\''):
+            context.validate()
+
+    def test_validate_with_invalid_user_locale_raises_exception(self):
+        context = parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': 'Android',
+                'browser_type': None,
+                'app_version': '1.0.0',
+                'user_locale': 'invalid',
+            },
+            {
+                'server_mode': 'dev',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid user locale \'invalid\''):
+            context.validate()
+
+    def test_validate_with_invalid_server_mode_raises_exception(self):
+        context = parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': 'Android',
+                'browser_type': None,
+                'app_version': '1.0.0',
+                'user_locale': 'en',
+            },
+            {
+                'server_mode': 'invalid',
+            },
+        )
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Invalid server mode \'invalid\''):
+            context.validate()
+
+
+class PlatformParameterFilterTests(test_utils.GenericTestBase):
+    """Test for the PlatformParameterFilter."""
+
+    def create_example_context(
+            self, client_type='Android', browser_type=None, app_version='1.2.3',
+            user_locale='en', mode='dev'):
+        """Creates and returns an EvaluationContext using the given
+        arguments.
+        """
+        return parameter_domain.EvaluationContext.from_dict(
+            {
+                'client_type': client_type,
+                'browser_type': browser_type,
+                'app_version': app_version,
+                'user_locale': user_locale,
+            },
+            {
+                'server_mode': mode,
+            },
+        )
+
+    def test_create_filter_from_dict(self):
+        filter_dict = {'type': 'app_version', 'conditions': [('=', '1.2.3')]}
+        filter_domain = (
+            parameter_domain
+            .PlatformParameterFilter.from_dict(filter_dict))
+
+        filter_domain.validate()
```

Comment:
Why validate if it’s not part of the test?

Reply:
Done, removed the validation.